atlasdb-tests-shared/src/main/java/com/palantir/timelock/paxos/InMemoryTimelockServices.java

```diff
@@ -176,13 +176,8 @@ public void close() {
         timeLockAgent.shutdown();
     }
 
-    public static InMemoryTimelockServices create(TemporaryFolder tempFolder) {
-        return create(tempFolder, "client");
-    }
-
-    public static InMemoryTimelockServices create(TemporaryFolder tempFolder, String client) {
+    static InMemoryTimelockServices create(TemporaryFolder tempFolder) {
         InMemoryTimelockServices services = new InMemoryTimelockServices(tempFolder);
-        services.setClient(client);
```

Comment:
it's a bit weird IMO how we have this `create` method just for the IMTST, and then we use the constructor for the actual use in the rule.

Reply:
yeah - this is because of the unfortunate way that InMemoryTimeLockRule needs to be set up so that it actually works:

- `InMemoryTimelockServices` must be initialised directly (not in the constructor), because it's used in the rule chain
- We initialise by calling `create`, because that method uses the tempFolder (which isn't guaranteed to be initialised yet)
- We need to set the client separately (and before `before`), otherwise it doesn't get used correctly

So we must use the constructor, so that gets called, then the tempFolder is initialised, then the client is set, and *then* we do the before() stuff.

Thankfully, I can recude the weirdness by switching IMTST to use the rule, after which point I can delete the create method.